### PR TITLE
[feat] Homepage UX Refresh Prototype

### LINKED ISSUE #1109: Integrate Recent Numbers section into home screen
**The product definition, design, content, and assets are a WIP and are neither finalized nor fully approved by WHO. The purpose of this task is to prototype the feature so we can execute quickly when the non-engineering deliverables are finalized.**

Should display current global daily cases and daily deaths
Should NOT include country or local

<details>
<summary>Rough mock (subject to change)</summary>
Screen Shot 2020-04-22 at 6 00 17 PM
</details>

### LINKED ISSUE #1110: Add home screen Information Card section
**The product definition, design, content, and assets are a WIP and are neither finalized nor fully approved by WHO. The purpose of this task is to prototype the feature so we can execute quickly when the non-engineering deliverables are finalized.**

* Dynamically loaded from a content bundle
* Content bundle contains multiple options for the section, UI chooses one at random (intention is for users to see different content when opening multiple times to make it feel new upon return)
* Can deep link to arbitrary page in the app

<details>
<summary>Rough mock (subject to change)</summary>
Screen Shot 2020-04-22 at 6 02 51 PM
</details>

### LINKED ISSUE #1111: Add Donate Now section to home screen
**The product definition, design, content, and assets are a WIP and are neither finalized nor fully approved by WHO. The purpose of this task is to prototype the feature so we can execute quickly when the non-engineering deliverables are finalized.**

Update UI to the new prototype

<details>
<summary>Rough mock (subject to change)</summary>
Screen Shot 2020-04-22 at 6 04 31 PM
</details>

### PR DESCRIPTION
**Pull request by hspinks**
Implements the new homepage prototype from the UX refresh designs.

There was an issue with caching content bundles loaded from the local asset bundler. Attempting to re-load a content bundle that did not exist from the local assets resulted in an exception being thrown that was not caught by our error handler [here](https://github.com/WorldHealthOrganization/app/blob/d33f87fc69a9c67c2adb3538a246eeec2165c257/client/flutter/lib/api/content/content_loading.dart#L52). As a non-ideal temporary workaround, I turned off caching when loading a local asset: https://github.com/WorldHealthOrganization/app/blob/d33f87fc69a9c67c2adb3538a246eeec2165c257/client/flutter/lib/api/content/content_loading.dart#L98 

Closes #1109 
Closes #1110 
Closes #1111 

#### Screenshots
<details>
<summary>Screenshots</summary>

Simulator Screen Shot - iPhone 11 Pro Max - 2020-05-01 at 16 21 56

Simulator Screen Shot - iPhone 11 Pro Max - 2020-05-01 at 16 22 04

Simulator Screen Shot - iPhone 11 Pro Max - 2020-05-01 at 16 22 08

</details>

#### How did you test the change?
* [x] iOS Simulator

---

#### Please follow this checklist. Please check each appropriate box (put an 'x' or check it after creating the PR).

- [x] REQUIRED: Do you have an Issue **assigned to you** for this PR?
- [x] Provided detailed pull request description and a succinct title
- [x] Tested your changes, especially after any code review iterations.
- [x] Included any relevant screenshots of UI updates.
- [x] Followed the [Contributor Guidelines](https://github.com/WorldHealthOrganization/app/blob/master/docs/CONTRIBUTING.md).
- [x] Verified all contributions are properly licensed pursuant to the [LICENSE](https://github.com/WorldHealthOrganization/app/blob/master/LICENSE) file in the root of the repository.
- [x] Verified your name is in the [content/credits.yaml](https://github.com/WorldHealthOrganization/app/blob/master/content/credits.yaml) file (if you want it to be).

### CODE REVIEW

File: client/flutter/lib/components/home_page_donate.dart [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  @override
+  Widget build(BuildContext context) {
+    return Stack(
+      children: <Widget>[
+        Positioned.fill(
+          child: _DonateBackground(),
+        ),
+        IntrinsicHeight(
+          child: Padding(
+            padding: EdgeInsets.fromLTRB(0.0, 72.0, 0.0, 60.0),
```
Comment (advayDev1) [inline client/flutter/lib/components/home_page_donate.dart:17]: if you are using only top/bottom, perhaps use .only

Comment (hspinks) [reply client/flutter/lib/components/home_page_donate.dart]: Originally, I was handling the padding differently to where I was using the sides, too - now that I am not, I will update 👍

File: client/flutter/lib/components/home_page_sections/home_page_donate.dart [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                    // TODO: localize
+                    'Help support the relief effort',
+                    variant: TypographyVariant.h2,
+                    style: TextStyle(
+                      color: CupertinoColors.white,
+                    ),
+                    textAlign: TextAlign.center,
+                  ),
+                ),
+                CupertinoButton(
```
Comment (advayDev1) [inline client/flutter/lib/components/home_page_sections/home_page_donate.dart:34]: PageButton for material/inkwell?

File: client/flutter/lib/components/home_page_sections/home_page_header.dart [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+import 'package:flutter/cupertino.dart';
+import 'package:flutter_svg/svg.dart';
+import 'package:who_app/components/themed_text.dart';
+import 'package:who_app/constants.dart';
+import 'package:who_app/generated/l10n.dart';
+
+class HomePageHeader extends StatelessWidget {
+  final HeaderType headerType;
+
+  HomePageHeader(this.headerType);
```
Comment (advayDev1) [inline client/flutter/lib/components/home_page_sections/home_page_header.dart:11]:
```suggestion
  const HomePageHeader(this.headerType);
```

File: client/flutter/lib/components/home_page_sections/home_page_information_card.dart [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            children: <Widget>[
+              ThemedText(
+                this.title,
+                variant: TypographyVariant.h3,
+                style: TextStyle(
+                  color: Constants.primaryDarkColor,
+                ),
+              ),
+              Container(
+                height: 6.0,
```
Comment (advayDev1) [inline client/flutter/lib/components/home_page_sections/home_page_information_card.dart:39]: 4 or 8?

File: client/flutter/lib/components/home_page_sections/home_page_recent_numbers.dart [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+import 'package:who_app/api/who_service.dart';
+import 'package:who_app/components/stat_card.dart';
+
+class HomePageRecentNumbers extends StatefulWidget {
+  @override
+  _HomePageRecentNumbersState createState() => _HomePageRecentNumbersState();
+}
+
+class _HomePageRecentNumbersState extends State<HomePageRecentNumbers> {
+  int globalCaseCount;
```
Comment (advayDev1) [inline client/flutter/lib/components/home_page_sections/home_page_recent_numbers.dart:12]:
```suggestion
  int _globalCaseCount;
```

File: client/flutter/lib/components/home_page_sections/home_page_recent_numbers.dart [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  int globalDeathCount;
```
Comment (advayDev1) [inline client/flutter/lib/components/home_page_sections/home_page_recent_numbers.dart:13]:
```suggestion
  int _globalDeathCount;
```

File: client/flutter/lib/components/home_page_sections/home_page_recent_numbers.dart [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+            // TODO: localize
+            title: 'Global Deaths',
+          ),
+        ],
+      ),
+    );
+  }
+
+  void fetchStats() async {
+    Map statsResponse = await WhoService.getCaseStats();
```
Comment (advayDev1) [inline client/flutter/lib/components/home_page_sections/home_page_recent_numbers.dart:53]:
```suggestion
    final statsResponse = await WhoService.getCaseStats();
```

we should get better at using final and automatic types everywhere

File: client/flutter/lib/components/home_page_sections/home_page_donate.dart (hunk 2 above)
Comment (hspinks) [reply client/flutter/lib/components/home_page_sections/home_page_donate.dart:34]: The design of this button is pretty different from the PageButton (see screenshot below) and doesn't have a background. @davidkaneda would you like this button to have an ink splash?

Screen Shot 2020-05-01 at 4 42 50 PM

File: client/flutter/lib/components/home_page_sections/home_page_information_card.dart (hunk 4 above)
Comment (hspinks) [reply client/flutter/lib/components/home_page_sections/home_page_information_card.dart:39]: I tried it and `6.0` looked best? 🤷‍♂️

File: client/flutter/lib/pages/main_pages/home_page.dart [hunk 8]
```diff
@@ -1,15 +1,131 @@
...
-        HomePageHeader(HeaderType.ProtectYourself),
+        _HomePageSection(
+          content: HomePageHeader(HeaderType.ProtectYourself),
+        ),
+        _HomePageSection(
+          padding: EdgeInsets.only(top: 56.0),
+          header: _HomePageSectionHeader(
+            // TODO: localize
+            title: 'Recent Numbers',
+            linkText: 'See all ›',
```
Comment (advayDev1) [inline client/flutter/lib/pages/main_pages/home_page.dart:28]: Do we want the carets inside the link as opposed to as an icon?

File: client/flutter/lib/pages/main_pages/home_page.dart [hunk 9]
```diff
@@ -1,15 +1,131 @@
...
+        alignment: WrapAlignment.spaceBetween,
+        crossAxisAlignment: WrapCrossAlignment.center,
+        spacing: 16.0,
+        children: <Widget>[
+          ThemedText(
+            this.title,
+            variant: TypographyVariant.h3,
+          ),
+          CupertinoButton(
+            padding: EdgeInsets.all(0.0),
```
Comment (advayDev1) [inline client/flutter/lib/pages/main_pages/home_page.dart:117]:
```suggestion
            padding: EdgeInsets.zero,
```

File: client/flutter/lib/pages/main_pages/routes.dart [hunk 10]
```diff
@@ -32,6 +33,7 @@ class Routes {
           dataSource: FactContent.getTheFacts,
           title: S.of(context).homePagePageButtonWHOMythBusters,
         ),
+    '/recent-numbers': (context) => RecentNumbersPage(),
```
Comment (advayDev1) [inline client/flutter/lib/pages/main_pages/routes.dart:36]: No action needed: We should check with David whether pushing this as opposed to selecting the tab is the right way to go for pages that also have tabs.  IMO Pushing is right, as you have it, just for simplicity.

File: client/flutter/lib/pages/main_pages/home_page.dart (hunk 8 above)
Comment (hspinks) [reply client/flutter/lib/pages/main_pages/home_page.dart:28]: This is definitely a good question... in the design, it just uses the character in text. When it comes to localization, for RTL we'll want to reverse the caret, which may mean it makes more sense to keep this in the localized string, or might mean that having it in the string makes it more difficult... I think we can decide that in a later PR

File: client/flutter/lib/components/home_page_sections/home_page_donate.dart (hunk 2 above)
Comment (advayDev1) [reply client/flutter/lib/components/home_page_sections/home_page_donate.dart:34]: Ah ok, its more  like a link than a button.  Nvm then!

**Pull request merged**